nautobot/extras/api/views.py

```diff
@@ -632,47 +659,65 @@ def approve(self, request, pk):
 
         return Response(serializer.data)
 
+    class ScheduledJobDeletePermissions(TokenPermissions):
+        """
+        As nautobot.core.api.authentication.TokenPermissions, but enforcing delete_scheduledjob not add_scheduledjob.
+        """
+
+        perms_map = {
+            "POST": ["%(app_label)s.delete_%(model_name)s"],
+        }
+
     @swagger_auto_schema(
         method="post",
         request_body=no_body,
     )
-    @action(detail=True, methods=["post"])
+    @action(detail=True, methods=["post"], permission_classes=[ScheduledJobDeletePermissions])
     def deny(self, request, pk):
-        if not request.user.has_perm("extras.run_job"):
-            raise PermissionDenied()
-
         scheduled_job = get_object_or_404(ScheduledJob, pk=pk)
 
+        if not (
+            Job.objects.check_perms(request.user, instance=scheduled_job.job_model, action="approve")
+            # A user can deny (revoke) their own approval requests, even without general job-approval permission
+            or request.user == scheduled_job.user
```

Comment:
I'm curious about this. I see it as allowing the user to pull back their own job request which seems the right but wonder if this might be a transparency concern in some environments.

Reply:
I thought there was existing behavior here around this but I'm not seeing it in `develop` at the moment. I wouldn't be opposed to removing it (both here and in the UI), as it would certainly simplify the logic in a few places.